template_files/handle_requests.jl

```diff
@@ -0,0 +1,99 @@
+using Downloads
+
+const AWS_LAMBDA_RUNTIME_API = ENV["AWS_LAMBDA_RUNTIME_API"]
+const HANDLER = ENV["_HANDLER"]  # from Dockerfile's CMD command
+const LAMBDA_TASK_ROOT = ENV["LAMBDA_TASK_ROOT"]
+const RUNTIME_URL = "http://$AWS_LAMBDA_RUNTIME_API/2018-06-01"
+
+abstract type State end
+struct Start <: State end
+struct Received <: State end
+struct Handled <: State end
+
+function error(::Start, exception, request_id)
+    @error "Unable to receive request" exception
+end
+
+function error(::Received, exception, request_id)
+    @error "Unable to handle event" exception request_id
+    post_error("runtime/invocation/$request_id/error", exception)
+end
+
+function error(::Handled, exception, request_id)
+    @error "Unable to notify lambda runtime (invocation response)" exception request_id
+end
+
+
+function http_get(url)
+    io = IOBuffer()
+    response = request(url; output=io)
+
+    seekstart(io)
+    body = read(io, String)
+
+    return response.headers, body
+end
+
+function http_post(url, headers, body)
+    io = IOBuffer()
+    write(io, body)
+    seekstart(io)  # We need to seekstart for the runtime to properly read the body
+
+    return request(url, method="POST", headers=headers, input=io)
+end
+
+function post_error(path::AbstractString, ex::Exception)
+    headers = [
+        "Content-type" => "application/json",
+        "Lambda-Runtime-Function-Error-Type" => "Unhandled",
```

Comment:
Could this be more informative?
```suggestion
        "Lambda-Runtime-Function-Error-Type" => typeof(ex),
```

According to the [Lambda docs](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-api.html):
```
Lambda-Runtime-Function-Error-Type – Error type that the extension encountered. Required: no.

This header consists of a string value. Lambda accepts any string, but we recommend a format of <category.reason>. For example:

Runtime.NoSuchHandler

Extension.APIKeyNotFound

Extension.ConfigInvalid

Extension.UnknownReason
```